### PR TITLE
feat: don't run 'dht_nice_tick()' when kadmelia is disabled

### PR DESCRIPTION
**Pull request by fabricedesre**

### CODE REVIEW

**Comment by fabricedesre** [comment]
@dignifiedquire the last commit moves the dht tick timer to its own `select!()` block that only runs when kadmelia is enabled. That changes a but the loop semantics when kadmelia is enabled but I think that's fine.

**Comment by dignifiedquire** [comment]
This should work: https://github.com/tokio-rs/tokio/issues/2583#issuecomment-638212772

**Comment by fabricedesre** [comment]
> This should work: [tokio-rs/tokio#2583 (comment)](https://github.com/tokio-rs/tokio/issues/2583#issuecomment-638212772)

so which commit do you prefer?

**Comment by dignifiedquire** [comment]
no strong preference, but the `OptionFuture` looks a little cleaner I think

**Comment by fabricedesre** [comment]
> no strong preference, but the `OptionFuture` looks a little cleaner I think

ok. I'm taking a look, but the loop and the fact that we don't deal with a single future but with `tick()` creating a new one for each call makes that not so straightforward.

File: iroh-p2p/src/node.rs [hunk 1]
```diff
@@ -169,7 +176,15 @@ impl<KeyStorage: Storage> Node<KeyStorage> {
...
-                _interval_event = nice_interval.tick() => {
+                _ = async {
+                    if let Some(ref mut nice_interval) = nice_interval {
+                        nice_interval.tick().await
+                    } else {
+                        // Will never be used, but needed to return the same type
+                        // as the `if ... {...}` branch above.
+                        time::Instant::now()
+                    }
+                }, if self.use_dht => {
```
[the remark this answers is not in the document]
Comment (fabricedesre) [reply iroh-p2p/src/node.rs]: done in latest commit

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**